### PR TITLE
Add basic micropython example.

### PR DESCRIPTION
**Pull request by AaronGlazer**
Example for https://github.com/KallistiOS/kos-ports/pull/30

Loads and runs a Python module from romdisk, then uses the micropython C API to call functions and inspect values from said module.

### CODE REVIEW

File: examples/dreamcast/micropython/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+clean: rm-elf
+	-rm -f $(OBJS)
+
+rm-elf:
+	-rm -f $(TARGET) romdisk.*
+
+$(TARGET): $(OBJS)
+	kos-cc -o $(TARGET) $(OBJS) -lmicropython
+
+romdisk.img:
```
Comment (andressbarajas) [inline examples/dreamcast/micropython/Makefile:26]: You can remove romdisk.img and romdisk.o from your makefile and stick 
`KOS_ROMDISK_DIR = romdisk`
 after OBJS += romdisk.o

File: examples/dreamcast/micropython/example.c [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+#include <micropython/py/lexer.h>
+#include <micropython/py/gc.h>
+#include <micropython/py/nlr.h>
+#include <micropython/py/compile.h>
+#include <micropython/py/runtime.h>
+#include <micropython/py/stackctrl.h>
+#include <micropython/py/qstr.h>
+#include <micropython/py/obj.h>
+
+extern uint8_t romdisk[];
```
Comment (andressbarajas) [inline examples/dreamcast/micropython/example.c:16]: With the latest KOS, these two romdisk lines can be removed

File: examples/dreamcast/micropython/Makefile (hunk 1 above)
Comment (andressbarajas) [reply examples/dreamcast/micropython/Makefile:26]: You can look at this file for an example:  https://github.com/KallistiOS/KallistiOS/blob/master/examples/dreamcast/vmu/vmu_game/Makefile

File: examples/dreamcast/micropython/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
+TARGET = micropython.elf
+
+CFLAGS += -std=c99
+CFLAGS += -I.
+CFLAGS += -Wall -Og
+
+SRCS += example.c
+OBJS += $(SRCS:.c=.o)
```
Comment (andressbarajas) [inline examples/dreamcast/micropython/Makefile:8]:
```suggestion
OBJS += example.o romdisk.o
```

File: examples/dreamcast/micropython/example.c [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        nlr_pop();
+    } else {
+        // uncaught exception
+        mp_obj_print_exception(&mp_plat_print, (mp_obj_t)nlr.ret_val);
+        exit(EXIT_FAILURE);
+    }
+}
+
+int main() {
```
Comment (gyrovorbis) [inline examples/dreamcast/micropython/example.c:37]:
```suggestion
int main(int argc, const char* argv[]) {
```

File: examples/dreamcast/micropython/Makefile (hunk 3 above)
Comment (gyrovorbis) [reply examples/dreamcast/micropython/Makefile:8]: I didn't think he needed this line anymore with `KOS_ROMDISK_DIR=romdisk`?

File: examples/dreamcast/micropython/example.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+#include <micropython/py/stackctrl.h>
+#include <micropython/py/qstr.h>
+#include <micropython/py/obj.h>
+
+static char mp_heap[8 * 1024];
+
+static void load_module(void) {
+    nlr_buf_t nlr;
+    if (nlr_push(&nlr) == 0) {
+        mp_lexer_t *lex = mp_lexer_new_from_file(qstr_from_str("/rd/script.py"));
```
Comment (AaronGlazer) [inline examples/dreamcast/micropython/example.c:21]: Leaving a comment here for myself for Falco's rec to add error checking

Comment (AaronGlazer) [reply examples/dreamcast/micropython/example.c]: @gyrovorbis I remembered that this `nlr` thing in `load_module` is some setjmp black magic to throw/catch errors so the whole demo is wrapped in it now and should catch any errors thrown by micropython.

File: examples/dreamcast/micropython/example.c [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        nlr_pop();
+    } else {
+        printf("demo ran into an uncaught exception!\n");
+        mp_obj_print_exception(&mp_plat_print, (mp_obj_t)nlr.ret_val);
+        exit(EXIT_FAILURE);
+    }
+
+    mp_deinit();
+
+    return 0;
```
Comment (gyrovorbis) [inline examples/dreamcast/micropython/example.c:81]:
```suggestion
    return EXIT_SUCCESS;
```

File: examples/dreamcast/micropython/Makefile [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# KallistiOS ##version##
+#
+# examples/dreamcast/micropython/Makefile
+# (c) 2023 Aaron Glazer
```
Comment (andressbarajas) [inline examples/dreamcast/micropython/Makefile:4]:
```suggestion
# Copyright (C) 2023 Aaron Glazer
```

File: examples/dreamcast/micropython/romdisk/script.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# KallistiOS ##version##
+#
+# script.py
+# (c) 2023 Aaron Glazer
```
Comment (andressbarajas) [inline examples/dreamcast/micropython/romdisk/script.py:4]:
```suggestion
# Copyright (C) 2023 Aaron Glazer
```

**Review by gyrovorbis — APPROVED** [review]
Looks good to me!

**Pull request merged**